sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/ParallelTransferOptions.java

```diff
@@ -63,30 +58,53 @@ public ProgressReceiver getProgressReceiver() {
      * @return The updated ParallelTransferOptions object.
      * @throws IllegalArgumentException when block size is less than 0 or greater than max blob block size (10MB).
      */
-    public ParallelTransferOptions setBlockSize(int blockSize) {
-        Utility.assertInBounds("blockSize", blockSize, 0, BLOB_MAX_BLOCK_SIZE);
+    public ParallelTransferOptions setBlockSize(Integer blockSize) {
+        if (blockSize != null) {
+            Utility.assertInBounds("blockSize", blockSize, 0, BLOB_MAX_UPLOAD_BLOCK_SIZE);
+        }
         this.blockSize = blockSize;
         return this;
     }
 
     /**
      * Sets the number of buffers being used for an upload/download operation.
      * @param numBuffers The number of buffers.
-     * For upload, The number of buffers is the maximum number of buffers this method should allocate.
+     * For buffered upload only, the number of buffers is the maximum number of buffers this method should allocate.
      * Must be at least two. Typically, the larger the number of buffers, the more parallel, and thus faster, the
      * upload portion  of this operation will be. The amount of memory consumed by this method may be up to
      * blockSize * numBuffers.
      * @return The updated ParallelTransferOptions object.
      * @throws IllegalArgumentException when numBuffers is less than 2.
      */
-    public ParallelTransferOptions setNumBuffers(int numBuffers) {
-        Utility.assertInBounds("numBuffers", numBuffers, 2, Integer.MAX_VALUE);
+    public ParallelTransferOptions setNumBuffers(Integer numBuffers) {
+        if (numBuffers != null) {
+            Utility.assertInBounds("numBuffers", numBuffers, 2, Integer.MAX_VALUE);
+        }
         this.numBuffers = numBuffers;
         return this;
     }
 
+    /**
+     * Sets the progress receiver for parallel reporting.
+     * @param progressReceiver The progress receiver.
+     * @return The updated ParallelTransferOptions object.
+     */
     public ParallelTransferOptions setProgressReceiver(ProgressReceiver progressReceiver) {
         this.progressReceiver = progressReceiver;
         return this;
     }
+
+    /**
+     * RESERVED FOR INTERNAL USE.
+     */
+    public void populateAndApplyDefaults(ParallelTransferOptions other) {
```

Comment:
This is a public API change. Does it need to be public? It would have to be reviewed with other language folks

Reply:
It has to be public because it's in a different package from where it's used

Reply:
Is this constructor used anywhere else other than BlobAsyncClient? If not could we continue to localize this there?

Reply:
It's BlobAsyncClient and BlobAsyncClientBase, which are in two different packages

Reply:
@conniey @alzimmermsft i am assuming this is resolved. I need this PR to go in.
/cc @pemari-msft fyi

Reply:
I've had an offline discussion on this today. I do not want us to get into the habit of creating 'reserved for internal use' public APIs that only exist to serve our needs in working around Java package limitations. I've suggested an alternative approach that should be considered, and @hemanttanwar is working through this for other APIs that exist solely to serve our needs in #5962. One option may be to push this as-is and then to follow-up in #5962 with it being made private again, but we should not ship with any API outside of the 'implementation' packages that is visible to users and reserved for internal use.

Reply:
doing it in #5962 sounds like a good option @JonathanGiles 
@rickle-msft please merge this PR

Reply:
Two comments:
1) This still needs to go through the escrow process.
2) I'm not convinced that this API will be able to be easily hidden using the *Helper approach, and would like to discuss in some more detail any other options that may exist to negate the need to introduce this method.